### PR TITLE
adding pydantic schema to prom_scrape

### PR DESCRIPTION
**Pull request by IbraAoad**
This PR implements the `pydantic` schema for the [Prometheus scrape interface](https://charmhub.io/prometheus-k8s/libraries/prometheus_scrape). Additionally, it corrects repos' URLs for providers and requirers.

### CODE REVIEW

File: interfaces/prometheus_scrape/v0/schema.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
+"""
+This file defines the schemas for the provider and requirer sides of the `prometheus_scrape` interface.
+
+It exposes two interfaces.schema_base.DataBagSchema subclasses called:
+- ProviderSchema
+- RequirerSchema
+
+Examples:
+    ProviderSchema:
+        application_data: {
```
Comment (PietroPasotti) [inline interfaces/prometheus_scrape/v0/schema.py:10]: this is a show-unit key, not in providerschema

File: interfaces/prometheus_scrape/v0/schema.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            ],
+            "scrape_metadata": {
+                "model": "cos",
+                "model_uuid": "c2e9f4d5-dcb3-4870-8509-330eb9745ee8",
+                "application": "zinc-k8s",
+                "unit": "zinc-k8s/0",
+                "charm_name": "zinc-k8s"
+            }
+        },
+        related_units: {
```
Comment (PietroPasotti) [inline interfaces/prometheus_scrape/v0/schema.py:51]: this is a show-unit key, not in providerschema

perhaps we should add a separate example for the RequirerSchema containing what's here?

File: interfaces/prometheus_scrape/v0/schema.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+class ScrapeJobModel(BaseModel):
+    job_name: Optional[str] = Field(
+        description="Name of the Prometheus scrape job, each job must be given a unique name",
+        default=None,
+    )
+    metrics_path: str = Field(description="Path for metrics scraping.")
+    static_configs: List[ScrapeStaticConfigModel]
+
+    class Config:
```
Comment (PietroPasotti) [inline interfaces/prometheus_scrape/v0/schema.py:110]: I'd put the model-level config on top of the class definition, not last.

```
class foo:
    class Config:
        ...
        
    baz: qux      
```

Comment (PietroPasotti) [reply interfaces/prometheus_scrape/v0/schema.py:110]: (also elsewhere)

File: interfaces/prometheus_scrape/v0/schema.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        extra = "allow"
+
+
+class ScrapeJobModel(BaseModel):
+    job_name: Optional[str] = Field(
+        description="Name of the Prometheus scrape job, each job must be given a unique name",
+        default=None,
+    )
+    metrics_path: str = Field(description="Path for metrics scraping.")
+    static_configs: List[ScrapeStaticConfigModel]
```
Comment (PietroPasotti) [inline interfaces/prometheus_scrape/v0/schema.py:108]: did you become tired of adding descriptions to all fields, or why does it stop here?

**Review by PietroPasotti — CHANGES_REQUESTED** [review]
Thanks! A few inconsistencies that need to be addressed, please :)

File: interfaces/prometheus_scrape/v0/schema.py (hunk 4 above)
Comment (IbraAoad) [reply interfaces/prometheus_scrape/v0/schema.py]: They felt pretty self-explanatory tbh that's why I haven't included them, Included all now tho :smile:

File: interfaces/prometheus_scrape/v0/schema.py (hunk 2 above)
Comment (PietroPasotti) [reply interfaces/prometheus_scrape/v0/schema.py:51]: ok so what confused me here is that in juju-status, the `related_units` key contains the 'other side' of the databag.

So I think it's fine if we replace this with:

```
unit:  {
                    "prometheus_scrape_unit_address": "zinc-k8s-0.zinc-k8s-endpoints.cos.svc.cluster.local",
                    "prometheus_scrape_unit_name": "zinc-k8s/0"
                }
    
```

**Review by PietroPasotti — APPROVED** [review]
Well done! Fine by me.

**Pull request merged**